auto_nag/multi_autofixers.py

```diff
@@ -0,0 +1,145 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import argparse
+from collections import ChainMap, defaultdict
+from typing import Any, Callable, Counter, Dict, List, Type
+
+from auto_nag import logger, utils
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.nag_me import Nag
+
+ToolsChanges = Dict[Type[BzCleaner], Any]
+
+
+class UnexpectedToolsError(Exception):
+    """Unexpected tools appear in merge function"""
+
+    def __init__(self, tools: List[Type[BzCleaner]]) -> None:
+        """Constructor
+
+        Args:
+            tools: the tools that change the same field.
+        """
+        super().__init__()
+        self.tools = tools
+
+    def __str__(self):
+        return (
+            "Error: merge function does not support merge fields "
+            f"from '{utils.english_list(self.tools)}'"
+        )
+
+
+class MissingMergeFunctionError(Exception):
+    """Merge function is missing"""
+
+    def __init__(self, field: str) -> None:
+        """Constructor
+
+        Args:
+            field: the field that is missing a merge function.
+        """
+        super().__init__()
+        self.field = field
+
+    def __str__(self):
+        return f"Error: missing merge function for '{self.field}'"
+
+
+class MultiAutoFixers:
+    "Merge changes from multiple tools and apply them to Bugzilla at once"
+
+    def __init__(
+        self, *tools: BzCleaner, **merge_functions: Callable[[ToolsChanges], dict]
+    ):
+        """Constructor
+
+        Args:
+            *tools: tools to merged their changes.
```

Comment:
typo + next line

Reply:
Fixed in d5ee660b1d9e6a3a04e603b54bfd570bc181c32c